### ISSUE TITLE
CapnProto format in return type

### ISSUE THREAD

**Issue by tobad357** [yandex/ClickHouse#2165]
We are trying to return responses as CapnProto struct but hitting some problems

test.capnp
```
@0x821d3cbafa234252

struct Test {
  id @0 :UInt64;
}
```

query
```
wget -O- -q "http://10.241.132.221:8123/?query=SELECT 1 AS id FORMAT CapnProto SETTINGS schema = 'test:Test'"
```

error
```
SELECT 1 AS id FORMAT CapnProto SETTINGS schema = 'test:Test' 
2018.04.03 15:23:09.176894 [ 49 ] <Error> HTTPHandler: Code: 62, e.displayText() = DB::Exception: Syntax error: failed at position 23: CapnProto SETTINGS schema = 'test:Test'
. Expected identifier, e.what() = DB::Exception, Stack trace:
```

Expected outcome would be a stream of bytes returned to the client
Are we doing something wrong or is there a bug?

**Comment by filimonov** [comment]
Unfortunately currently CapnProto is supported only as input format:
https://github.com/yandex/ClickHouse/blob/master/dbms/src/DataStreams/FormatFactory.cpp

**Comment by tobad357** [comment]
That would explain it
It seems like the documentation is a bit eager
https://clickhouse.yandex/docs/en/formats/capnproto/

**Comment by ztlpn** [comment]
@filimonov is right, but the particular error that you get is a parsing error. FORMAT clause should be after the SETTINGS clause.

**Comment by filimonov** [comment]
> It seems like the documentation is a bit eager

Yep. May be @vavrusa was planning to add that as an output format too? It doesn't sound like a big issue.

**Comment by tobad357** [comment]
It would be pretty great to be able to get CapnProto or other structured formats back from the DB
That way integration with any server side would be very straight forward

**Comment by vavrusa** [comment]
I didn't have any immediate plans, as the Kafka engine doesn't support writing yet. I have some plans to add support for that so that ClickHouse can be used as a stream processor reading/write to Kafka, but probably late April/May as I'm a bit busy with other work stuff. I can help reviewing/helping out with PRs though!
